### DISCUSSION TITLE
Agregados

### DISCUSSION THREAD

**Discussion by JanainaPaula** [balta-io/2810#4] [Q&A] [answered]
Olá! 
Não consegui entender muito bem o que são agregados e qual é a diferença de bounded contexts? São eventos relacionados a um Dominio? Por exemplo, um evento chamado pedido_canceled é um agregado dos eventos relacionados ao domínio de pedido?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @JanainaPaula , como vai?

Agragados são entidades que em conjunto fazem parte de um agregado raiz, que é indicado como o objeto principal de um contexto.
Por exemplo, em um contexto de pedidos, a entidade pedido po ser classificada como agregada raiz, sendo que sem ela o pedido não existe. Da mesma forma, o cliente é um agregado, uma entidade que compoe um pedido e assim por diante.

Os bounded contexts ou contextos delimitados são recortes que fazemos no domínio e que na maioria das vezes reflete uma capacidade de negócio. É complicado mesmo dar exemplo de bounded context pois eles podem variar muito, mas um exemplo que sempre tento utilizar é o de compras que fazemos nos cursos.

Temos um contexto para pedidos, outro para backoffice, outro para storefront e assim por dia... eles são a delimitação, a segmentação do "O que vai... onde fica"
